packages/web-components/src/search-widget/server/utils/map-minimal-properties.ts

```diff
@@ -0,0 +1,34 @@
+import pickBy from 'lodash.pickby'
+import { PagedResultPropertyModel_, PagedResultPropertyImageModel_ } from '@reapit/foundations-ts-definitions'
+import {
+  PickedPagedResultPropertyModel_,
+  PickedPagedResultPropertyImageModel_,
+  PickedPropertyModel,
+  PickedPropertyImageModel,
+} from '../../types'
+
+type InputObject = PagedResultPropertyModel_ | PagedResultPropertyImageModel_
+type PickedModel = PickedPropertyModel | PickedPropertyImageModel
+type OutputObject = PickedPagedResultPropertyModel_ | PickedPagedResultPropertyImageModel_
+
+/**
+ * Map through to return only necessary properties
+ */
+export const mapMinimalProperties = (
+  { _embedded, ...rest }: InputObject,
+  includedEmbeddedProps: string[],
+): OutputObject => {
+  if (!_embedded) {
+    return {
+      ...rest,
+      _embedded,
```

Comment:
I think this `_embedded` will be `undefined` or `null`. So should it be `_embedded: []` ?

Reply:
No, if `_embedded `is undefined, then it should return the original object so that it will be compatible with type from `@reapit/foundations-ts-definitions`